electron/libelectron/src/backends/memory_mgmt.cc

```diff
@@ -0,0 +1,110 @@
+#include "memory_mgmt.hh"
+namespace memory_mgmt{
+
+FirstFitAllocator::FirstFitAllocator(size_t dev_mem_size) {
+
+    /* Initialize structures*/
+    device_memory = mmap(NULL,dev_mem_size,PROT_EXEC|PROT_READ|PROT_WRITE,MAP_ANONYMOUS|MAP_SHARED,0,0);
+    heap_start=std::make_shared<struct MemBlock>();
+    heap_start->size = dev_mem_size;
+    heap_start->used = 0;
+    heap_start->ptr = device_memory;
+
+    heap_start->prev = heap_start;
+    heap_start->next = heap_start;
```

Comment:
Still not sure here -- do we need this to be a circular linked list? I think keeping the tail pointing to null might make things a bit less weird

Reply:
Updated it to normal linked list